### ISSUE TITLE
opencv_webcam example is doing to much work on the main thread

### ISSUE THREAD

**Issue by rodja** [zauberzeug/nicegui#490]
The example works as expected but is doing its work on the main thread. If developers take this example and use it in their code it will produce sluggish UI responses.

**Comment by rodja** [comment]
The [image capturing](https://github.com/zauberzeug/nicegui/blob/6d97db659cf80b99076b0fa1f145ae6a2e9d5561/examples/opencv_webcam/main.py#L23) is IO bound and could be "asyncified" with NiceGUIs `background_task.create`.

The [image encoding](https://github.com/zauberzeug/nicegui/blob/6d97db659cf80b99076b0fa1f145ae6a2e9d5561/examples/opencv_webcam/main.py#L26-L27) needs to run in a separate process because its CPU intensive and would only block the GIL when done in a background task.

**Comment by falkoschindler** [comment]
- [ ] Tests for 3.7 and 3.8 are read, because the `cancel_futures` parameter has only been added in Python 3.9.
- [ ] Furthermore, when I try to stop the app with Ctrl-C, I get an exception and the process keeps running:
    > concurrent.futures.process.BrokenProcessPool: A process in the process pool was terminated abruptly while the future was running or pending.

**Comment by rodja** [comment]
That was **not easy**. When using subprocesses we need to make sure they are stopped on shutdown. Otherwise they will block the procedure and will leave you with a non-working, unresponsive app. But the internals of the subprocesses are complaining if they are killed like I did in 4fd0791. So it's better to ensure all subprocesses can complete and no new ones are scheduled before the actual shutdown is performed. I've now implemented this by [disconnecting all clients on shutdown](https://github.com/zauberzeug/nicegui/blob/5ddf26549aea6de3a828aca4a0b4aaf95f429975/examples/opencv_webcam/main.py#L74). If no new images are requested, there will be no new subprocesses started.
Additionally when using Ctrl-C to abort the app from the console, the connection within the multiprocessing queue is complaining about KeyboardInterrupts. It's nicer if these are hidden or not occurring at all. So we must prevent scheduling of new subprocesses **before** Uvicorn goes into shutdown via SIGINT. I've done so by registering a signal handler to first disconnect and after a 1 second delay let Uvicorn do it's part: https://github.com/zauberzeug/nicegui/blob/de333afbf3b38a1b1b6b73cecc5b3b9378f526e1/examples/opencv_webcam/main.py#L64-L68
